nighthawk/source/client/stream_decoder.cc

```diff
@@ -0,0 +1,68 @@
+#include "nighthawk/source/client/stream_decoder.h"
+
+#include <memory>
+
+#include "common/http/http1/codec_impl.h"
+#include "common/http/utility.h"
+
+namespace Nighthawk {
+namespace Client {
+
+void StreamDecoder::decodeHeaders(Envoy::Http::HeaderMapPtr&& headers, bool end_stream) {
+  ASSERT(!complete_);
+  complete_ = end_stream;
+  response_headers_ = std::move(headers);
```

Comment:
FWIW, you will also need to track `response_trailers_` and look at `grpc-status` when doing gRPC load tests later on. It's fine not to do this for now, maybe just leave a TODO.